Mathlib/FieldTheory/PurelyInseparable/Basic.lean

```diff
@@ -591,3 +591,53 @@ theorem adjoin_eq_of_isAlgebraic [Algebra.IsAlgebraic F E] :
   simp only [S, coe_map, IsScalarTower.coe_toAlgHom', IntermediateField.algebraMap_apply]
 
 end separableClosure
+
+section
+
+open TensorProduct
+
+variable {k K R : Type*} [Field k] [Field K] [Algebra k K] [CommRing R] [Algebra k R]
+
+lemma IsPurelyInseparable.exists_pow_pow_mem_range_tensorProduct_of_expChar
+    [IsPurelyInseparable k K] (q : ℕ) [ExpChar k q] (x : R ⊗[k] K) :
+    ∃ n, x ^ q ^ n ∈ (algebraMap R (R ⊗[k] K)).range := by
+  cases subsingleton_or_nontrivial (R ⊗[k] K)
+  · exact ⟨0, 0, Subsingleton.elim _ _⟩
+  obtain (hq|hq) := expChar_is_prime_or_one k q
+  induction x with
+  | zero => exact ⟨0, 0, by simp [zero_pow_eq, hq.ne_zero]⟩
+  | add x y hx hy =>
+    obtain ⟨n, hx⟩ := hx
+    obtain ⟨m, hy⟩ := hy
+    use n + m
+    have : ExpChar (R ⊗[k] K) q := by
+      refine expChar_of_injective_ringHom
+        (f := Algebra.TensorProduct.includeRight.toRingHom.comp (algebraMap k K)) ?_ q
+      apply RingHom.injective
+    rw [add_pow_expChar_pow, pow_add]
+    nth_rw 2 [mul_comm]
+    rw [pow_mul, pow_mul]
+    exact Subring.add_mem _ (Subring.pow_mem _ hx _) (Subring.pow_mem _ hy _)
```

Comment:
I've extracted `Subalgebra.perfectClosure` from this, and you might want to merge [this](https://github.com/leanprover-community/mathlib4/commit/e2b810d70934c878b930a27741922a89a449b1e4) as well.